src/dispatcher.rs

```diff
@@ -172,8 +198,13 @@ impl TxnDispatcher for Dispatcher {
         self.batch_handler.end_batch(&mut ctx);
 
         let (tags, roothash_messages) = ctx.close();
-        (outputs, tags, roothash_messages)
+        Ok((outputs, tags, roothash_messages))
     }
 
     fn finalize(&self, _new_storage_root: Hash) {}
+
+    /// Configure abort batch flag.
+    fn set_abort_batch_flag(&mut self, abort_batch: Arc<AtomicBool>) {
```

Comment:
This method isn't used anywhere. Also, the argument seems suspicious. Namely, that a second thread that has a clone of the original `abort_batch_flag` will never see that another thread aborted the batch. Just to confirm, it's one thread per batch, right?

Reply:
I'm not 100% sure what you're asking, but yeah there is one thread that is processing the batches (spawned here: https://github.com/oasisprotocol/oasis-core/blob/master/runtime/src/dispatcher.rs#L135-L138), but this flag is initialized & set (and watched) from a different thread: see: https://github.com/oasisprotocol/oasis-core/blob/master/runtime/src/dispatcher.rs#L131